.eslintrc.json

```diff
@@ -1,8 +1,112 @@
 {
-  "extends": ["next/core-web-vitals", "prettier"],
-  "plugins": ["prettier"],
-  "ignorePatterns": ["**/dist/*.js", "/docs/**", "/out/**"],
+  "plugins": ["prettier", "@typescript-eslint", "import"],
+  "parserOptions": {
+    "project": "./tsconfig.json"
+  },
+  "parser": "@typescript-eslint/parser",
+  "extends": [
+    "plugin:@typescript-eslint/strict-type-checked",
+    "plugin:@typescript-eslint/stylistic-type-checked",
+    "plugin:import/recommended",
+    "plugin:import/typescript",
+    "next/core-web-vitals",
+    "prettier"
+  ],
+  "ignorePatterns": [
+    "**/dist/*",
+    "/docs/**",
+    "/out/**",
+    "**/jest.config.js",
+    "**/next.config.js",
+    "**/postcss.config.js",
+    "**/tailwind.config.js"
+  ],
   "rules": {
-    "prettier/prettier": "error"
-  }
+    "prettier/prettier": "error",
+    "import/order": [
+      "warn",
+      {
+        "groups": ["builtin", "external", "internal", ["parent", "sibling"]],
+        "pathGroupsExcludedImportTypes": ["builtin"],
+        "newlines-between": "never",
+        "alphabetize": {
+          "order": "asc",
+          "caseInsensitive": true
+        }
+      }
+    ],
+    "import/no-empty-named-blocks": "error",
+    "import/no-mutable-exports": "error",
+    "import/no-cycle": "error",
```

Comment:
Prevents import cycles - they can produce extremely hard to debug and reproduce bugs sometimes